Source/WebCore/bindings/scripts/CodeGeneratorJS.pm

```diff
@@ -6965,11 +6965,16 @@ sub GenerateCallbackImplementationContent
             if ($operation->extendedAttributes->{RethrowException}) {
                 push(@$contentRef, "        auto throwScope = DECLARE_THROW_SCOPE(vm);\n");
                 push(@$contentRef, "        throwException(&lexicalGlobalObject, throwScope, returnedException);\n");
+                push(@$contentRef, "        return CallbackResultType::ExceptionThrown;\n");
+            } elsif ($operation->extendedAttributes->{TreatExceptionAsRejectedPromise}) {
```

Comment:
I think we should match the spec instead:
https://webidl.spec.whatwg.org/#js-invoking-callback-functions

We shouldn't need a new `TreatExceptionAsRejectedPromise` non-standard IDL extended attribute. The spec says that if the callback's return type is a promise and an exception is thrown, we should reject the promise with this exception. So we should conditionalize your new behavior on whether or not the return type is a promise, NOT on a new `TreatExceptionAsRejectedPromise` non-standard IDL extended attribute I think. 

Hopefully, it's not that much more work and it would improve the spec compliance of our generated bindings.

Reply:
Ah nice, I did not realise it was specced like that. I agree, nice to be compatible with the spec here.